### PR TITLE
Redesign deposit and withdraw blocks for UX

### LINKED ISSUE #141: Redesign deposit and withdraw blocks for UX
Deposit & Withdraw operation related to `NeoFS chain balance`, but not good UX for understanding it.

image

 It might be deposit/withdraw icons in `NeoFS chain balance` block.

### PR DESCRIPTION
**Pull request by mike-petrov**
Signed-off-by: Mikhail Petrov <mike@nspcc.ru>

closes #141

### CODE REVIEW

**Comment by mike-petrov** [comment]
image

File: src/App.js [hunk 1]
```diff
@@ -525,6 +537,89 @@ export const App = () => {
...
+		if (depositQuantity * 1e8 >= 1 && depositQuantity * 1e8 <= neoBalanceTemp) {
+			onModal('approveRequest');
+			const invocations = [{
+				scriptHash: NeoFSContract.gasToken,
+				operation: 'transfer',
+				args: [
+					{ type: 'Hash160', value: Neon.create.account(walletData.account.address).scriptHash },
+					{ type: 'Hash160', value: Neon.create.account(NeoFSContract.account).scriptHash },
+					{ type: 'Integer', value: depositQuantity * 1e8 },
+					{ type: 'ByteArray', value: '' },
```
Comment (roman-khimov) [inline src/App.js:550]: type: Any
value: null

File: src/App.js [hunk 2]
```diff
@@ -525,6 +537,89 @@ export const App = () => {
...
+						onModal('failed', 'Something went wrong, try again');
+					}
+				});
+			}
+			if (!response.error) {
+				setDepositQuantity(0);
+				onModal('success', response.txid ? response.txid : response);
+			}
+		} else {
+			onPopup('failed', 'Incorrect quantity value');
```
Comment (roman-khimov) [inline src/App.js:582]: Incorrect amount

File: src/App.js [hunk 3]
```diff
@@ -525,6 +537,89 @@ export const App = () => {
...
+				response = await dapi.invoke({ ...invocations[0], signers }).catch((err) => handleError(err));
+			} else {
+				response = await wcSdk.invokeFunction({ invocations, signers }).catch((err) => handleError(err));
+			}
+			if (response && !response.message) {
+				setWithdrawQuantity(0);
+				onModal('success', response.txid ? response.txid : response);
+			}
+		} else {
+			onPopup('failed', 'Incorrect quantity value');
```
Comment (roman-khimov) [inline src/App.js:619]: Incorrect amount

File: src/App.js [hunk 4]
```diff
@@ -525,6 +537,89 @@ export const App = () => {
 		}
 	};
 
+	const onDeposit = async (neoBalanceTemp) => {
+		if (depositQuantity * 1e8 >= 1 && depositQuantity * 1e8 <= neoBalanceTemp) {
```
Comment (roman-khimov) [inline src/App.js:541]: Can it be fractional (1.5 GAS)? We accept it here.

File: src/App.js [hunk 5]
```diff
@@ -525,6 +537,89 @@ export const App = () => {
...
+				setDepositQuantity(0);
+				onModal('success', response.txid ? response.txid : response);
+			}
+		} else {
+			onPopup('failed', 'Incorrect quantity value');
+		}
+	};
+
+	const onWithdraw = async (neoFSBalanceTemp) => {
+		if (withdrawQuantity >= 1 && withdrawQuantity * 1e12 <= neoFSBalanceTemp) {
```
Comment (roman-khimov) [inline src/App.js:587]: Because this one can't, we need some integer amount of GAS.

File: src/App.js (hunk 4 above)
Comment (mike-petrov) [reply src/App.js:541]: yes, it can

File: src/App.js (hunk 5 above)
Comment (mike-petrov) [reply src/App.js:587]: in this case, it can't, because the input with onKeyPress function, which allows only integer numbers

**Comment by roman-khimov** [comment]
Squash 'em?

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**